internal/setup/setup.go

```diff
@@ -0,0 +1,195 @@
+package setup
+
+import (
+	"fmt"
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/nestoca/joy/internal/config"
+	"github.com/nestoca/joy/internal/dependencies"
+	"github.com/nestoca/joy/internal/style"
+	"github.com/nestoca/joy/pkg/catalog"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+)
+
+const defaultCatalogDir = "~/.joy"
+
+func Setup(configDir, catalogDir, catalogRepo string) error {
```

Comment:
@j-martin what are your opinions on mega functions like this? 

I think breaking this function down into smaller chunks, called by the Setup() function would make this more readable/maintainable.

Reply:
Thanks for raising that, I definitely missed it. I also think it is an issue. 

From the (quick) look at it, the comments on the method could make good methods if extracted.

Reply:
Idk I agree with @silphid that specifically in Go [mega](https://github.com/golang/go/blob/9cdcb01320d9a866e46a2daedb9bde16e0d51278/src/fmt/print.go#L766) [functions](https://github.com/golang/go/blob/9cdcb01320d9a866e46a2daedb9bde16e0d51278/src/fmt/print.go#L681) [are](https://github.com/golang/go/blob/9cdcb01320d9a866e46a2daedb9bde16e0d51278/src/strconv/atoi.go#L78) [fairly](https://github.com/golang/go/blob/9cdcb01320d9a866e46a2daedb9bde16e0d51278/src/strconv/decimal.go#L110) [common](https://github.com/golang/go/blob/9cdcb01320d9a866e46a2daedb9bde16e0d51278/src/time/format.go#L184).

It should be decided on a case by case basis, if it makes sense it makes sense because sometimes Go is hella procedural and breaking into multiple smaller functions just makes everything super verbose (e.g. passing around all kinds of errors and signals).

I think at this point we should stop bikeshedding this PR and if we wanna refactor we make a new PR with a proposed refactoring and let's move on.

Reply:
The linked method is less than 100 lines long, but yes it is already too long. :P ;)

Reply:
I linked 5 methods, the first one is well over 100 lines long 😉  but whatever I always knew you had Ruby design patterns in your heart 😜

Reply:
I linked 5 methods, the first one is well over 100 lines long 😉  but whatever I always knew you had [Ruby design patterns](https://github.com/rubocop/ruby-style-guide#short-methods) in your heart 😜